src/rust/engine/process_execution/src/remote.rs

```diff
@@ -2,25 +2,36 @@ use std::error::Error;
 use std::sync::Arc;
 
 use bazel_protos;
+use boxfuture::{BoxFuture, Boxable};
 use digest::{Digest, FixedOutput};
+use futures::{Future, future};
 use grpcio;
 use protobuf::{self, Message, ProtobufEnum};
 use sha2::Sha256;
 
 use super::{ExecuteProcessRequest, ExecuteProcessResult};
 
 pub struct CommandRunner {
-  execution_client: bazel_protos::remote_execution_grpc::ExecutionClient,
-  operations_client: bazel_protos::operations_grpc::OperationsClient,
+  execution_client: Arc<bazel_protos::remote_execution_grpc::ExecutionClient>,
+  operations_client: Arc<bazel_protos::operations_grpc::OperationsClient>,
+}
+
+#[derive(Debug, Eq, PartialEq)]
```

Comment:
Equality matching on an error is an interesting precedent... I think I'd rather see the equivalent of a match to type-test (rather than ever inspecting the string) and then have more error variants? Probably this is just for testing, but still...

Reply:
Turns out this isn't actually used any more at all (was previously used in testing). Removed.